src/app/app-components/authors-input-component/authors-input-component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, ViewChild, ElementRef, forwardRef, Input, HostBinding, HostListener } from '@angular/core';
+import { FormControl, ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { MatAutocomplete, MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { Observable } from 'rxjs';
+import { map, filter } from 'rxjs/operators';
+import { Author } from '@app/app-models';
+
+@Component({
+  selector: 'app-authors-input-component',
+  templateUrl: './authors-input-component.html',
+  styleUrls: ['./authors-input-component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => AuthorsInputComponent),
+      multi: true
+    },
+  ]
+})
+export class AuthorsInputComponent implements ControlValueAccessor {
+  @Input()
+  options: Author[] = [];
+  // Allow the input to be disabled, and when it is make it somewhat transparent.
+  @Input() disabled = false;
+  @HostBinding('style.opacity')
+  get opacity() {
+    return this.disabled ? 0.25 : 1;
+  }
+
+  @HostListener('click', ['$event'])
+  handleHostClick(event: MouseEvent) {
+    this.onTouched();
+  }
+  readonly authorControl = new FormControl();
+  readonly filteredAuthors$: Observable<Author[]>;
+  addOnBlur = true;
+  authors: Author[] = [];
+
+  @ViewChild('authorInput', { static: false }) authorInput: ElementRef<HTMLInputElement>;
+  @ViewChild('autocomplete', { static: false }) matAutocomplete: MatAutocomplete;
+
+  // Function to call when the length changes.
+  onChange = (authors: Author[]) => { };
+  // Function to call when the input is touched.
+  onTouched = () => { };
+
+  constructor() {
+    this.filteredAuthors$ = this.authorControl.valueChanges.pipe(
+      filter(author => typeof author === typeof String()),
```

Comment:
We could use `typeof author === 'string'`.

Depending on the Code Style Guide in use, type classes like String, Number, Boolean may be forbidden.

Reply:
But, here you're not using the `new String()`, so it should be fine.